win/WIN_ToggleButton.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include "WIN_UIelement.h"
+#include <SDL.h>
+#include <vector>
+#include <memory>
+
+namespace win
+{
+	class ToggleButton;
+	typedef void (*ToggleActionFunction)(ToggleButton* button);
```

Comment:
Potentially in WIN_UIelement.h (or a new WIN_Action.h)

`typedef void (*ActionFunction)(UIelement * control)`

and then Button and ToggleButton can share the definition at the cost of a `static_cast<Button *>(control)` or `static_cast<ToggleButton *>(control)` in the action functions themselves.
**COULD**

Reply:
done